fuel-vm/src/arith.rs

```diff
@@ -30,55 +28,3 @@ pub fn checked_add_usize(a: usize, b: usize) -> Result<usize, PanicReason> {
 pub fn checked_sub_usize(a: usize, b: usize) -> Result<usize, PanicReason> {
     a.checked_sub(b).ok_or(PanicReason::ArithmeticOverflow)
 }
-
-/// Add two checked words. Might wrap if overflow
-///
-/// The function is inlined so it will be optimized to `a + b` if `optimized` feature is
-/// enabled so the operation is unchecked.
-///
-/// This should be used in contexts that are checked and guaranteed by the protocol to
-/// never overflow, but then they might due to some bug in the code.
-#[inline(always)]
-pub fn add_word(a: Word, b: Word) -> Result<Word, RuntimeError> {
```

Comment:
Is this code change necessary to support the `no_std` work, or is it simply tangential?

Reply:
Not striclty necessary, but the error/bug type rework is needed. It would be more work to keep it like this, and it needs fixing anyways.

Reply:
Yeah, I definitely agree that it is helpful work towards the main goal - I simply ask because sometimes we can separate discrete chunks of work into separate PRs. But that comes down to personal preference